functions/HttpTriggerAPIUsersIdTaskId/__init__.py

```diff
@@ -6,71 +10,124 @@
 #GET API method function
 def get(param1, param2):
     #connects to db
+    logging.debug('Attempting a db conenction')
     cnxn = connect()
-    cursor = cnxn.cursor()
-    logging.info('opened connection')        
-    logging.info('Going to execute select query')
+    cursor = cnxn.cursor() 
     try:
+        logging.debug('opened connection')        
+        logging.debug('Attempting to execute GET task query')
         #Get task title, description and user name by userId and taskId
-        sql_query = ("""SELECT tasks.title, tasks.description, CONCAT (users.firstName, ' ', users.lastName) AS "user" 
-        FROM [dbo].[tasks] JOIN [dbo].[users] 
-        on [dbo].[tasks].userId = [dbo].[users].userId
-        WHERE [dbo].[users].userId = ? AND [dbo].[tasks].taskId = ?""")
+        sql_query = ("""SELECT tasks.title, tasks.description, tasks.createdDate, tasks.dueDate, 
+                    tasks.completed, tasks.completedDate, CONCAT (users.firstName, ' ', users.lastName) AS "user" 
+                    FROM [dbo].[tasks] JOIN [dbo].[users] 
+                    on [dbo].[tasks].userId = [dbo].[users].userId
+                    WHERE [dbo].[users].userId = ? AND [dbo].[tasks].taskId = ?""")
         cursor.execute(sql_query, param1, param2)
-        logging.info('Executed the query')         
-        data = cursor.fetchall()
-        logging.info(f"Got result: {data}")
-        return data
-#irrespective of results, closes connection
+        logging.debug('Executed the query')          
+        row = cursor.fetchone()
+        logging.info(f"Got result: {row}")
+        if not row:
+            logging.error('No record with the requested parameters')
+            return func.HttpResponse('Task not found', status_code=500)

```

Comment:
I think this should actually be a 404 status code because the resource wasn't found. It's not really an error on our side per se.

Reply:
updated